scipy/optimize/_shuffled_complex_evolution.py

```diff
@@ -0,0 +1,1083 @@
+            return self.func(x, *self.args, **self.kwargs)
+        elif (self.nargs > 0) and (self.nkwargs == 0):
+            return self.func(x, *self.args)
+        elif (self.nargs == 0) and (self.nkwargs > 0):
+            return self.func(x, **self.kwargs)
+        else:
+            return self.func(x)
+
+
+# from distutils.util (Python 3.11.1)
+def _strtobool(val):
+    """
+    Convert a string representation of truth to true (1) or false (0).
+    True values are 'y', 'yes', 't', 'true', 'on', and '1'; false values
+    are 'n', 'no', 'f', 'false', 'off', and '0'.  Raises ValueError if
+    'val' is anything else.
+
+    """
+    val = val.lower()
+    if val in ('y', 'yes', 't', 'true', 'on', '1'):
+        return 1
+    elif val in ('n', 'no', 'f', 'false', 'off', '0'):
+        return 0
+    else:
+        raise ValueError("invalid truth value {!r}".format(val))
+
+
+def shuffled_complex_evolution(
+        func, x0, lb, ub=None,
+        mask=None,
+        args=(), kwargs={},
+        sampling='half-open',
+        maxn=1000, kstop=10, pcento=0.0001, peps=0.001,
+        ngs=2, npg=0, nps=0, nspl=0, mings=0,
+        seed=None, iniflg=True,
+        alpha=0.8, beta=0.45, maxit=False, printit=2,
+        polish=True,
+        restart=False, restartfile1='',
+        restartfile2=''):
+    """
+    Shuffled Complex Evolution algorithm for finding the minimum of a
+    multivariate function
+
+    The SCE or SCE-UA method is a general purpose global optimization, which
+    can be used with high-dimensional problems. It was used successfully,
+    for example, to calibrate hydrologic models with more than 50 parameters
+    [5]_.
+    The algorithm has been described in detail in Duan et al. [1]_ and [2]_.
+    Another paper of Duan et al. [3]_ discusses how to use the method
+    effectively. The implementation here also includes the recommendations of
+    Behrangi et al. [4]_.
+
+    Parameters
+    ----------
+    func : callable
+        Function in the form ``func(x, *args, **kwargs)``, where ``x`` are
+        the parameters in the form of an iterable.
+        ``args`` and ``kwargs`` are passed to the function via the usual
+        unpacking operators.
+    x0 : array_like
+        Parameter values with `mask==1` used in initial complex if
+        `iniflg==1`.
+    lb : array_like, sequence or `Bounds`
+        Lower bounds of parameters if ``ub`` given.
+        If `ub` is not given, then `lb` are the bounds of the parameters,
+        either 1) as an instance of the `Bounds` class, or 2) as ``(min, max)``
+        pairs for each element in ``x``, defining the finite lower and upper
+        bounds for the parameters of `func`.
+    ub : array_like, optional
+        Upper bounds of parameters.
+    mask : array_like, optional
+        Include (1, True) or exclude (0, False) parameters in minimization
+        (default: include all parameters). The number of parameters ``nopt`` is
+        ``sum(mask)``.
```

Comment:
```suggestion
```

This is a big API change compared to scipy's other optimizers that do not expose such functionality. It is a useful feature in principle but needs to be discussed on a bigger scale first.